### ISSUE TITLE
Simplify build setup

### ISSUE THREAD

**Issue by mhalbritter** [spring-projects/spring-grpc#82]
Right now, users have to deal with various plugins and their configuration when using Spring gRPC (see the configuration generated on start.spring.io).

I'm not sure if it's technically possible, but maybe a Spring gRPC Maven and Gradle plugin could solve that problem.

That way, users only have to apply one plugin instead of multiple and dealing with their config.

**Comment by dsyer** [comment]
Nice idea. But I hate the thought of maintaining and publishing Maven and Gradle plugins that only do things that are already available elsewhere (albeit possibly with better defaults). For Maven wouldn't it be better just to include some default configuration in the parent? For Gradle I wouldn't know where to start, but there must be analogues (e.g. native image compilation - we don't have our own plugin for that do we).

**Comment by mhalbritter** [comment]
> For Maven wouldn't it be better just to include some default configuration in the parent?

Yes. But as most projects have Spring Boot as their parent, you're limited to scope import, which doesn't support plugin configuration. Maven 4 fixes that I think.

> For Gradle I wouldn't know where to start, but there must be analogues (e.g. native image compilation - we don't have our own plugin for that do we).

The Spring Boot plugin reacts to the application of the native plugin and configures the build.

**Comment by mhalbritter** [comment]
The spring-grpc Gradle plugin could apply the `com.google.protobuf` plugin and configure it.

I'm not sure if it's possible, but the Maven plugin could add and configure the `os-maven-plugin` and the `protobuf-maven-plugin` (to work around the Maven limitations). No need to duplicate the existing plugins.

**Comment by dsyer** [comment]
I'm happy to wait and see if Spring Boot takes these responsibilities, so no need to do anything here for now.

**Comment by onobc** [comment]
@mhalbritter are you ok w/ us closing this issue for now since we are not going to do any of this plugin work in Spring gRPC as of now. We can revisit once we integrate into Spring Boot 4.0.x.
